### ISSUE TITLE
Split `user` table into local and federated data

### ISSUE THREAD

**Issue by Nutomic** [LemmyNet/lemmy#1360]
At the moment we are storing everything related to the user in the same table, things like email, password, federation keypair, preferences. And we store both local and remote users in that table, which means that remote users also have fields for password or settings, which makes no sense.

We should split this table into a local part (email, password, settings, ...), and a federated part (displayname, keypair, inbox url, ...). The same also applies to the community. Maybe we could even have an `actor` table that is used both for user and community data, as their federation data is quite similar.

**Comment by dessalines** [comment]
My current idea is to :

- Rename the `user_` table to `person` (the official activitypub name for a user actor. `actor` wouldn't be good because there are also group actors).
- create a new table named `local_user`, that contains local user info only, including settings. I went through the current `User_` table and figure out its these columns. 

table | columns
--- | ---
person | 15
local_user | 12

```
pub struct User_ {
  pub id: i32, // person
  pub name: String, // person
  pub preferred_username: Option<String>, // person
  pub password_encrypted: String, // local_user
  pub email: Option<String>, // local_user
  pub avatar: Option<String>, //person
  pub admin: bool, // local_user
  pub banned: bool, // person?
  pub published: chrono::NaiveDateTime, // person
  pub updated: Option<chrono::NaiveDateTime>, // person
  pub show_nsfw: bool, // local_user
  pub theme: String, // local_user
  pub default_sort_type: i16, // local_user
  pub default_listing_type: i16, // local_user
  pub lang: String, // local_user
  pub show_avatars: bool, // local_user
  pub send_notifications_to_email: bool, // local_user
  pub matrix_user_id: Option<String>, // local_user
  pub actor_id: Url, // person
  pub bio: Option<String>, // person
  pub local: bool, // person
  pub private_key: Option<String>, // person
  pub public_key: Option<String>, // person
  pub last_refreshed_at: chrono::NaiveDateTime, // person
  pub banner: Option<String>, // person
  pub deleted: bool, // local_user
  pub inbox_url: Url, // person
  pub shared_inbox_url: Option<Url>, // person
}
```

It doesn't make sense to split out "settings", because many of the federated fields like `bio`, `name`, etc are actor fields.

**Comment by Nutomic** [comment]
I think the federated fields for community are (almost) identical. So we could instead make a table `actor`, and store federation data for both users and communities there. But maybe its better to have separate tables `person` and `group`, not sure.

**Comment by dessalines** [comment]
I just looked over the community table, for now it doesn't make sense to break it out. I think there's only two fields that differentiate a local community from a federated one.

**Comment by dessalines** [comment]
Fixed by #1428
